baseplate/experiments/targeting/tree_targeting.py

```diff
@@ -0,0 +1,192 @@
+        """Build a node to evaluate equality (single value or list).
+
+        :param dict input_node: dict with the field name and value or list
+        of values to accept for targeting. This dict will contain two keys:
+        "field", and one of "value" or "values". If "value" is provided,
+        a single value is expected. If "values" is instead provided, then
+        a list of values is expected.
+
+        A full EqualNode in a targeting tree configuration looks like this:
+        {
+            EQ:{
+                field: <field_name>
+                value: <accepted_value>
+            }
+        }
+
+        The expected input to this constructor from the above example would be:
+        {
+            field: <field_name>,
+            value: <accepted_value>
+        }
+        """
+
+        if len(input_node) != 2:
+            raise ValueError("EqualNode expects exactly two fields.")
+
+        if not 'field' in input_node:
+            raise ValueError("EqualNode expects input key 'field'.")
+
+        if not 'value' in input_node and not 'values' in input_node:
+            raise ValueError("EqualNode expects input key 'value' or 'values'.")
+
+        self._accepted_key = input_node.get('field').lower()
+        self._accepted_values = input_node.get('values') or [input_node.get('value')]
+
+    def evaluate(self, **kwargs):
+        candidate_value = kwargs.get(self._accepted_key)
+        if candidate_value in self._accepted_values:
+            return True
+
+        return False
+
+
+class AllNode(Targeting):
+    """All child nodes return True"""
+    def __init__(self, input_node):
+        """Build a node to evaluate multiple children and return True if they
+        all evaluate to True (boolean 'and').
+
+        :param list input_node: a list of Targeting nodes
+        """
+        if not isinstance(input_node, list):
+            raise TypeError("Input to AllNode expects a list.")
+
+        self._children = []
+
+        for node in input_node:
+            self._children.append(create_targeting_tree(node))
+
+    def evaluate(self, **kwargs):
+        for node in self._children:
+            if not node.evaluate(**kwargs):
+                return False
+
+        return True
+
+
+class AnyNode(Targeting):
+    """At least one child node return True"""
+
+    def __init__(self, input_node):
+        """Build a node to evaluate multiple children and return True if at
+        least one of them evaluates to True (boolean 'or').
+
+        :param list input_node: a list of Targeting nodes
+        """
+        if not isinstance(input_node, list):
+            raise TypeError("Input to AnyNode expects a list.")
+
+        self._children = []
+
+        for node in input_node:
+            self._children.append(create_targeting_tree(node))
+
+    def evaluate(self, **kwargs):
+        for node in self._children:
+            if node.evaluate(**kwargs):
+                return True
+
+        return False
```

Comment:
Like `all`, python has an `any` method, you can do this with `return any(node.evaluate(**kwargs) for node in self._children)`